common/crypto/attestation-api/test/attested_evidence_test.sh

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# *** README ***
+# This script is meant to run as part of the build.
+# The script is transferred to the folder where other test binaries will be located,
+# and it will orchestrate the test.
+# Orchestration involves: preparing input file for init_attestation,
+# calling get_attestation, calling attestation_to_evidence, calling verify_evidence.
+
+set -e
+
+. ../scripts/attestation_to_evidence.sh
+. ../scripts/define_to_variable.sh
+. ../scripts/enclave_to_mrenclave.sh
+
+DEFINES_FILEPATH="${FPC_PATH}/common/crypto/attestation-api/test/common/test-defines.h"
+TAGS_FILEPATH="${FPC_PATH}/common/crypto/attestation-api/attestation/attestation_tags.h"
+
+function remove_artifacts()
+{
+    rm -rf *.txt
+}
+
+function orchestrate()
+{
+    #get attestation
+    ./get_attestation_app
+    define_to_variable "${DEFINES_FILEPATH}" "GET_ATTESTATION_OUTPUT"
+    [ -f ${GET_ATTESTATION_OUTPUT} ] || die "no output from get_attestation"
+
+    #translate attestation
+    ATTESTATION=$(cat ${GET_ATTESTATION_OUTPUT})
+    attestation_to_evidence "${ATTESTATION}"
+
+    define_to_variable "${DEFINES_FILEPATH}" "EVIDENCE_FILE"
+    echo ${EVIDENCE} > ${EVIDENCE_FILE}
+
+    #verify evidence
+    ./verify_evidence_app
+}
+
+#######################################
+# sim mode test
+#######################################
+
+say "Testing simulated attestation"
+
+#prepare input
+remove_artifacts
+define_to_variable "${DEFINES_FILEPATH}" "CODE_ID_FILE"
+define_to_variable "${DEFINES_FILEPATH}" "STATEMENT_FILE"
+define_to_variable "${DEFINES_FILEPATH}" "STATEMENT"
+define_to_variable "${DEFINES_FILEPATH}" "INIT_DATA_INPUT"
+
+define_to_variable "${TAGS_FILEPATH}" "ATTESTATION_TYPE_TAG"
+define_to_variable "${TAGS_FILEPATH}" "SIMULATED_TYPE_TAG"
+
+echo -n "this is ignored" > ${CODE_ID_FILE}
+echo -n "also ignored" > ${STATEMENT_FILE}
+echo -n "{\"${ATTESTATION_TYPE_TAG}\": \"${SIMULATED_TYPE_TAG}\"}" > ${INIT_DATA_INPUT}
+
+orchestrate
+
+say "Test simulated attestation success"
+
+#######################################
+# hw mode test
+#######################################
+if [[ ${SGX_MODE} == "HW" ]]; then
+    say "Testing HW-mode attestation"
+
+    #prepare input
+    remove_artifacts
+    define_to_variable "${DEFINES_FILEPATH}" "UNSIGNED_ENCLAVE_FILENAME"
+    enclave_to_mrenclave ${UNSIGNED_ENCLAVE_FILENAME} test_enclave.config.xml
+    echo -n "$MRENCLAVE" > ${CODE_ID_FILE}
+    echo -n ${STATEMENT} > ${STATEMENT_FILE}
+
+    #get spid type
+    SPID_TYPE_FILEPATH="${FPC_PATH}/config/ias/spid_type.txt"
```

Comment:
this is nice and should also be added to readme.

Reply:
good catch, this is a breaking update!
done